powerline/segments/i3wm.py

```diff
@@ -39,27 +48,58 @@ def workspaces(pl, segment_info, only_show=None, output=None, strip=0):
 
 	Highlight groups used: ``workspace`` or ``w_visible``, ``workspace`` or ``w_focused``, ``workspace`` or ``w_urgent``.
 	'''
-	global conn
-	if not conn:
-		try:
-			import i3ipc
-		except ImportError:
-			import i3 as conn
-		else:
-			conn = i3ipc.Connection()
-
 	output = output or segment_info.get('output')
 
 	return [{
-		'contents': w['name'][min(len(w['name']), strip):],
+		'contents': w['name'][strip:],
 		'highlight_groups': calcgrp(w)
-	} for w in conn.get_workspaces()
-		if (not only_show or any(w[typ] for typ in only_show))
-		and (not output or w['output'] == output)
+	} for w in get_i3_connection().get_workspaces()
+		if ((not only_show or any(w[typ] for typ in only_show))
+		     and (not output or w['output'] == output))
 	]
 
 
 @requires_segment_info
+def workspace(pl, segment_info, workspace=None, strip=0):
+	'''Return state for the specified workspace
+
+	:param str workspace:
+		Specifies which workspace to show. If unspecified, may be set by the 
+		``list_workspaces`` lister if used, otherwise falls back to 
+		currently focused workspace.
+
+	:param int strip:
```

Comment:
actually they differ:

    def workspaces(pl, segment_info, only_show=None, output=None, strip=0):
        	'''Return list of used workspaces

vs

    def workspace(pl, segment_info, workspace=None, strip=False):
        	'''Return state for the specified workspace